src/main/java/org/opentripplanner/index/IndexAPI.java

```diff
@@ -355,17 +357,52 @@ public Response getRoute (@PathParam("routeId") String routeIdString) {
    /** Return all stop patterns used by trips on the given route. */
    @GET
    @Path("/routes/{routeId}/patterns")
-   public Response getPatternsForRoute (@PathParam("routeId") String routeIdString) {
+   public Response getPatternsForRoute (@PathParam("routeId") String routeIdString,
+                                        @QueryParam("includeGeometry") boolean includeGeometry) {
        FeedScopedId routeId = GtfsLibrary.convertIdFromString(routeIdString);
        Route route = index.routeForId.get(routeId);
        if (route != null) {
            Collection<TripPattern> patterns = index.patternsForRoute.get(route);
-           return Response.status(Status.OK).entity(PatternShort.list(patterns)).build();
+           List<EncodedPolylineBean> patternGeometries = patterns
+                   .stream()
+                   .map(pattern -> PolylineEncoder.createEncodings(pattern.geometry))
+                   .collect(Collectors.toList());
+
+
+           List<PatternShort> output;
+
+           if (includeGeometry) {
+            output = PatternShort.list(new ArrayList<>(patterns), patternGeometries);
```

Comment:
Might as well include all of this within the if block as follows:

```suggestion
           List<PatternShort> output;

           if (includeGeometry) {
               List<EncodedPolylineBean> patternGeometries = patterns
                   .stream()
                   .map(pattern -> PolylineEncoder.createEncodings(pattern.geometry))
                   .collect(Collectors.toList());
               output = PatternShort.list(new ArrayList<>(patterns), patternGeometries);
```

Reply:
This is much cleaner thanks! Refactored in 1360e72e44678135324c29f7ac4942514fa05270